### ISSUE TITLE
[BUG] iOS freeze on startup

### ISSUE THREAD

**Issue by albilaga** [reactiveui/Akavache#550]
**Describe the bug**
In iOS when want to get value from `BlobCacbe` in startup it always freeze or if we are using await or subscribe iOS is always faster to open first page. It become a problem when we want to check value from`BlobCache` and decide which page to opened first

**Steps To Reproduce**
1. Open https://github.com/albilaga/akavachetest
2. Run for iOS
3. It always freeze on startup and can't continue.

**Expected behavior**
It should can get value from `BlobCache`

**Environment**
- OS:      iOS
- Device:  test on iPhone Simulator
- Version:   Akavache 6.9.10

**Comment by PureWeen** [comment]
From what I remember it's a little tricky depending on async things during startup. If you use `Result` that's going to most likely lock a thread and freeze the application.

Someone might know a clever way to do this with Akavache but you might look into using a simpler storage point to pivot on for startup. Something like

https://docs.microsoft.com/en-us/xamarin/essentials/preferences

It's a synchronous API that just grabs a quick value from local storage. You could use that to store where you will want a user to go and then use akavache for any additional data you want loaded.

**Comment by albilaga** [comment]
Yeah we change that to using `Preferences` to store `IsLogin` state. But this means we create another property that we actually can rely on if `AccessToken` is empty or not. And we can't save `AccessToken` on `Preferences` for that.

**Comment by EmilAlipiev** [comment]
I think that i have posted quite detailed question somewhere regarding that. Akavache doesnt work on App start. You can use Xamarin.Essentials Preferences as you said. thats what i am doing. Akavache more convenient for complex json objects

**Comment by EmilAlipiev** [comment]
just found it here https://github.com/reactiveui/Akavache/issues/392

**Comment by manijak** [comment]
Hi there, didn't want to start a new issue but I'm seeing this as well on Xamarin.Mac. In my case I cannot use Essentials as I need to fetch/store a blob (file), among other things. 

Is there any way to make this work? I need to fetch some files when the applications starts. But every time it calls GetObject it freezes. If I call the same code in a button click event, it works just fine. 

Thanks

**Comment by glennawatson** [comment]
Thank you for your report. Closing as the initialization deadlock has been resolved in Akavache V11.

V10's `BlobCache` static constructor could deadlock on iOS when accessed from the main thread during startup. The static constructor triggered SQLite initialization, which scheduled work on a background thread and then blocked waiting for it — a classic main-thread deadlock.

V11 eliminated this:

- No static constructors — initialization is explicit via `CacheDatabase.Initialize<T>()`
- SQLite provider initialization (`Batteries_V2.Init()`) is called synchronously during builder configuration, not deferred to a background thread
- All subsequent cache operations use standard async/await through sqlite-net-pcl's `SQLiteAsyncConnection`

```csharp
// Call during app startup — no deadlock risk
CacheDatabase.Initialize<SystemJsonSerializer>(builder =>
    builder.WithApplicationName("MyApp")
           .WithSqliteProvider()
           .WithSqliteDefaults());

var value = await CacheDatabase.UserAccount.GetObject<string>("key");
```

If you experience freezes with V11, please open a new issue.
